File: src/types/projectNote.ts
/** 案件ノート（Rust `ProjectNote` の TS 側ミラー）。フィールド名は snake_case のまま保持する。 */
export interface ProjectNote {
  project_id: string;
  user_md: string;
  ai_md: string | null;
  ai_edited: boolean;
  ai_generated_at: string | null;
  updated_at: string | null;
}

/** AI要約の世代管理履歴（Rust `AiHistoryEntry` の TS 側ミラー）。 */
export interface AiHistoryEntry {
  id: string;
  project_id: string;
  ai_md: string;
  replaced_at: string;
}

/** AI要約生成コマンドの戻り値（Rust `GenerateNoteOutcome` の TS 側ミラー）。 */
export interface GenerateNoteOutcome {
  ai_md: string;
  dropped_mails: number;
}

File: src-tauri/src/models/project_note.rs
use serde::{Deserialize, Serialize};

/// 案件ノート。正本は DB 側（PIGEON-CONTEXT.md はディレクトリ連携時のミラー）。
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ProjectNote {
    pub project_id: String,
    /// 「ノート」タブ: ユーザー手書き（Markdown/GFM）
    pub user_md: String,
    /// 「AI要約」タブ: AI が生成した下書き。ユーザー編集可
    pub ai_md: Option<String>,
    /// ユーザーが ai_md を手修正したか（再生成時の確認ダイアログ判定に使う）
    pub ai_edited: bool,
    pub ai_generated_at: Option<String>,
    pub updated_at: Option<String>,
}

/// AI要約の再生成履歴。
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AiHistoryEntry {
    pub id: String,
    pub project_id: String,
    pub ai_md: String,
    pub replaced_at: String,
}
